source/InfiniteLoader/InfiniteLoader.js

```diff
@@ -134,11 +134,9 @@ export default class InfiniteLoader extends React.PureComponent {
     });
 
     // For memoize comparison
-    const squashedUnloadedRanges = unloadedRanges.reduce(
-      (reduced, unloadedRange) =>
-        reduced.concat([unloadedRange.startIndex, unloadedRange.stopIndex]),
-      [],
-    );
+    const squashedUnloadedRanges = [].concat(...unloadedRanges.map(
```

Comment:
I'm confused why can't we just do 

```
const squashedUnloadedRanges = unloadedRanges.map(....
```

Reply:
A little-known feature of `concat` is that it flattens top-level array arguments:
```
> [].concat([1,2], [3,4])
(4) [1, 2, 3, 4]
```
`unloadedRanges.map(...)` would return an array of arrays rather than a flattened array.

Reply:
Oh I see. TIL.

Reply:
Depending on unlimited varargs is not a good idea

Reply:
Is that true even when using `Array.prototype.concat.apply([], unloadedRanges)`?

Reply:
@aptfx also, do you have a link to documentation of this?

Reply:
Yup nevermind, PR coming in a moment